### ISSUE TITLE
v 0.45.1 MSSQL not working after upgrade

### ISSUE THREAD

**Issue by ejconquer85** [metabase/metabase#27699]
**Describe the bug**
After upgrading to version 0.45.1 all the MSSQL queries stopped working.  If I try to create a new one also doesn't work. It's a Docker installation on AWS EC2.

**DockerFile**
```dockerfile
FROM metabase/metabase:v0.45.1

ENV MB_DB_CONNECTION_TIMEOUT_MS=60000
ENV MB_PASSWORD_COMPLEXITY=normal
ENV MB_PASSWORD_LENGTH=8
ENV MB_JETTY_PORT=3000
ENV MB_DB_TYPE=postgres
ENV MB_DB_DBNAME=MetabaseDB
ENV MB_DB_PORT=5432
ENV MB_EMOJI_IN_LOGS=false
ENV MB_JETTY_ASYNC_RESPONSE_TIMEOUT=6000000

COPY ./plugins/sparksql-databricks.metabase-driver.jar /plugins/
COPY ./plugins/athena.metabase-driver.jar /plugins/

RUN chmod 744 /plugins/sparksql-databricks.metabase-driver.jar
RUN chmod 744 /plugins/athena.metabase-driver.jar
```

**Logs**
[94df38f7-0bca-45af-9d79-c2ed5055e524] 2023-01-16T15:03:53+01:00 ERROR metabase.query-processor.middleware.catch-exceptions Error processing query: Connections could not be acquired from the underlying database!
{:database_id 3,
 :started_at #t "2023-01-16T14:03:25.666341Z[GMT]",
 :via
 [{:status :failed,
   :class com.mchange.v2.resourcepool.CannotAcquireResourceException,
   :error "A ResourcePool could not acquire a resource from its primary factory or source.",
   :stacktrace
   ["com.mchange.v2.resourcepool.BasicResourcePool.awaitAvailable(BasicResourcePool.java:1507)"
    "com.mchange.v2.resourcepool.BasicResourcePool.prelimCheckoutResource(BasicResourcePool.java:644)"
    "com.mchange.v2.resourcepool.BasicResourcePool.checkoutResource(BasicResourcePool.java:554)"
    "com.mchange.v2.c3p0.impl.C3P0PooledConnectionPool.checkoutAndMarkConnectionInUse(C3P0PooledConnectionPool.java:758)"
    "com.mchange.v2.c3p0.impl.C3P0PooledConnectionPool.checkoutPooledConnection(C3P0PooledConnectionPool.java:685)"
    "com.mchange.v2.c3p0.impl.AbstractPoolBackedDataSource.getConnection(AbstractPoolBackedDataSource.java:140)"
    "--> driver.sql_jdbc.execute$fn__55821.invokeStatic(execute.clj:191)"
    "driver.sql_jdbc.execute$fn__55821.invoke(execute.clj:189)"
    "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:496)"
    "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:485)"
    "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:493)"
    "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:485)"
    "driver.sql_jdbc$fn__88395.invokeStatic(sql_jdbc.clj:58)"
    "driver.sql_jdbc$fn__88395.invoke(sql_jdbc.clj:56)"
    "query_processor.context$executef.invokeStatic(context.clj:59)"
    "query_processor.context$executef.invoke(context.clj:48)"
    "query_processor.context.default$default_runf.invokeStatic(default.clj:67)"
    "query_processor.context.default$default_runf.invoke(default.clj:65)"
    "query_processor.context$runf.invokeStatic(context.clj:45)"
    "query_processor.context$runf.invoke(context.clj:39)"
    "query_processor.reducible$identity_qp.invokeStatic(reducible.clj:12)"
    "query_processor.reducible$identity_qp.invoke(reducible.clj:9)"
    "query_processor.middleware.cache$run_query_with_cache.invokeStatic(cache.clj:193)"
    "query_processor.middleware.cache$run_query_with_cache.invoke(cache.clj:178)"
    "query_processor.middleware.cache$maybe_return_cached_results$maybe_return_cached_results_STAR___53529.invoke(cache.clj:219)"
    "query_processor.middleware.permissions$check_query_permissions$fn__49184.invoke(permissions.clj:109)"
    "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__52474.invoke(mbql_to_native.clj:23)"
    "query_processor$fn__55005$combined_post_process__55010$combined_post_process_STAR___55011.invoke(query_processor.clj:212)"
    "query_processor$fn__55005$combined_pre_process__55006$combined_pre_process_STAR___55007.invoke(query_processor.clj:209)"
    "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__53450$fn__53455.invoke(resolve_database_and_driver.clj:35)"
    "driver$do_with_driver.invokeStatic(driver.clj:76)"
    "driver$do_with_driver.invoke(driver.clj:72)"
    "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__53450.invoke(resolve_database_and_driver.clj:34)"
    "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__49450.invoke(fetch_source_query.clj:314)"
    "query_processor.middleware.store$initialize_store$fn__49640$fn__49641.invoke(store.clj:11)"
    "query_processor.store$do_with_store.invokeStatic(store.clj:45)"
    "query_processor.store$do_with_store.invoke(store.clj:39)"
    "query_processor.middleware.store$initialize_store$fn__49640.invoke(store.clj:10)"
    "query_processor.middleware.normalize_query$normalize$fn__53722.invoke(normalize_query.clj:22)"
    "query_processor.middleware.constraints$add_default_userland_constraints$fn__50728.invoke(constraints.clj:53)"
    "query_processor.middleware.process_userland_query$process_userland_query$fn__53661.invoke(process_userland_query.clj:145)"
    "query_processor.middleware.catch_exceptions$catch_exceptions$fn__54033.invoke(catch_exceptions.clj:167)"
    "query_processor.reducible$async_qp$qp_STAR___45482$thunk__45484.invoke(reducible.clj:100)"
    "query_processor.reducible$async_qp$qp_STAR___45482$fn__45486.invoke(reducible.clj:105)"]}
  {:status :failed,
   :class java.sql.SQLException,
   :error "Connections could not be acquired from the underlying database!",
   :stacktrace
   ["com.mchange.v2.sql.SqlUtils.toSQLException(SqlUtils.java:118)"
    "com.mchange.v2.c3p0.impl.C3P0PooledConnectionPool.checkoutPooledConnection(C3P0PooledConnectionPool.java:692)"
    "com.mchange.v2.c3p0.impl.AbstractPoolBackedDataSource.getConnection(AbstractPoolBackedDataSource.java:140)"
    "--> driver.sql_jdbc.execute$fn__55821.invokeStatic(execute.clj:191)"
    "driver.sql_jdbc.execute$fn__55821.invoke(execute.clj:189)"
    "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:496)"
    "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:485)"
    "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:493)"
    "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:485)"
    "driver.sql_jdbc$fn__88395.invokeStatic(sql_jdbc.clj:58)"
    "driver.sql_jdbc$fn__88395.invoke(sql_jdbc.clj:56)"
    "query_processor.context$executef.invokeStatic(context.clj:59)"
    "query_processor.context$executef.invoke(context.clj:48)"
    "query_processor.context.default$default_runf.invokeStatic(default.clj:67)"
    "query_processor.context.default$default_runf.invoke(default.clj:65)"
    "query_processor.context$runf.invokeStatic(context.clj:45)"
    "query_processor.context$runf.invoke(context.clj:39)"
    "query_processor.reducible$identity_qp.invokeStatic(reducible.clj:12)"
    "query_processor.reducible$identity_qp.invoke(reducible.clj:9)"
    "query_processor.middleware.cache$run_query_with_cache.invokeStatic(cache.clj:193)"
    "query_processor.middleware.cache$run_query_with_cache.invoke(cache.clj:178)"
    "query_processor.middleware.cache$maybe_return_cached_results$maybe_return_cached_results_STAR___53529.invoke(cache.clj:219)"
    "query_processor.middleware.permissions$check_query_permissions$fn__49184.invoke(permissions.clj:109)"
    "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__52474.invoke(mbql_to_native.clj:23)"
    "query_processor$fn__55005$combined_post_process__55010$combined_post_process_STAR___55011.invoke(query_processor.clj:212)"
    "query_processor$fn__55005$combined_pre_process__55006$combined_pre_process_STAR___55007.invoke(query_processor.clj:209)"
    "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__53450$fn__53455.invoke(resolve_database_and_driver.clj:35)"
    "driver$do_with_driver.invokeStatic(driver.clj:76)"
    "driver$do_with_driver.invoke(driver.clj:72)"
    "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__53450.invoke(resolve_database_and_driver.clj:34)"
    "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__49450.invoke(fetch_source_query.clj:314)"
    "query_processor.middleware.store$initialize_store$fn__49640$fn__49641.invoke(store.clj:11)"
    "query_processor.store$do_with_store.invokeStatic(store.clj:45)"
    "query_processor.store$do_with_store.invoke(store.clj:39)"
    "query_processor.middleware.store$initialize_store$fn__49640.invoke(store.clj:10)"
    "query_processor.middleware.normalize_query$normalize$fn__53722.invoke(normalize_query.clj:22)"
    "query_processor.middleware.constraints$add_default_userland_constraints$fn__50728.invoke(constraints.clj:53)"
    "query_processor.middleware.process_userland_query$process_userland_query$fn__53661.invoke(process_userland_query.clj:145)"
    "query_processor.middleware.catch_exceptions$catch_exceptions$fn__54033.invoke(catch_exceptions.clj:167)"
    "query_processor.reducible$async_qp$qp_STAR___45482$thunk__45484.invoke(reducible.clj:100)"
    "query_processor.reducible$async_qp$qp_STAR___45482$fn__45486.invoke(reducible.clj:105)"],
   :state nil}],
 :json_query
 {:constraints {:max-results 10000, :max-results-bare-rows 2000},
  :type :native,
  :middleware {:js-int-to-string? true, :ignore-cached-results? false, :process-viz-settings? false},
  :native
  {:query
   "(my query is logged here)",
   :template-tags {}},
  :database 3,
  :parameters [],
  :async? true,
  :cache-ttl 1249},
 :status :failed,
 :class java.lang.NullPointerException,
 :stacktrace [nil],
 :card_id 211,
 :context :question,
 :error "Connections could not be acquired from the underlying database!",
 :row_count 0,
 :running_time 0,
 :data {:rows [], :cols []}}

**To Reproduce**
Execute any card or query in MSSQL

**Severity**
We can't see any MSSQL report, MySQL and DataBricks doesn't have any problem.

**Metabase Diagnostic Info**

```json
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.17+8",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.17",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.17+8",
    "os.name": "Linux",
    "os.version": "4.14.301-224.520.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "sparksql-databricks",
      "sqlserver",
      "mysql",
      "mongo"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "14.3"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.5.0"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2022-12-07",
      "tag": "v0.45.1",
      "branch": "release-x.45.x",
      "hash": "019d31c"
    },
    "settings": {
      "report-timezone": "Etc/GMT"
    }
  }
}
```

**Comment by paoliniluis** [comment]
@ejconquer85 Can you specify which MS SQL version so we can test it?

**Comment by ejconquer85** [comment]
Any MSSQL (I have tried Standard 2017 and WebEdition 2019 ) database is failing, the issue appears to be data in the Metabase database after the upgrade. I changed the Metabase connection in my server to a new clean one and did a quick configuration as if it were a new instance, after that I configured a MSSQL connection (the same that is failing in my prod installation) and it worked. I changed back to my prod DB and the bug is there again. So probably is some configuration data in the Metabase DB. I can't create new MSSQL connections or use the ones that I already had.
